flow/connectors/clickhouse/qrep_avro_sync.go

```diff
@@ -255,3 +255,24 @@ func (s *ClickHouseAvroSyncMethod) writeToAvroFile(
 
 	return avroFile, nil
 }
+
+func castToDestinationTypeIfNeeded(avroColumnName string, columnName string, columnSettings []*protos.ColumnSetting) string {
+	if len(columnSettings) == 0 {
+		return fmt.Sprintf("`%s`", avroColumnName)
+	}
+
+	destinationType := ""
+	for _, col := range columnSettings {
+		if col.SourceName == columnName {
+			destinationType = col.DestinationType
+		}
+	}
+
+	// TODO: check destination type is different from source type
+	switch destinationType {
+	case "String", "Nullable(String)":
+		return fmt.Sprintf("CAST(%s, '%s')", avroColumnName, destinationType)
```

Comment:
```suggestion
		return fmt.Sprintf("CAST(`%s`, '%s')", avroColumnName, EscapeStr(destinationType))
```

Reply:
oops good catch

Reply:
Can column names contain backticks as well?

Reply:
hmmm yeah interesting! looks like it's possible in postgres with double quotes[[ref](https://www.postgresql.org/docs/7.0/syntax525.htm)]
```
postgres=# create table t1 ("a`" int);
CREATE TABLE
postgres=# insert into t1 ("a`") values (1);
INSERT 0 1
postgres=# select * from t1;
 a`
----
  1
(1 row)
```
but is not possible in Avro [[ref](https://avro.apache.org/docs/1.8.1/spec.html#names)]. so i suspect we already performed some kind of conversion here.